cpp/include/cudf/copying.hpp

```diff
@@ -627,10 +627,25 @@ std::unique_ptr<table> boolean_mask_scatter(
  *
  * @returns Returns a table by scattering `input` into `target` as per `boolean_mask`.
  */
- std::unique_ptr<table> boolean_mask_scatter(
+std::unique_ptr<table> boolean_mask_scatter(
     std::vector<std::reference_wrapper<scalar>> const& input, table_view const& target,
     column_view const& boolean_mask,
     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Get the element at specified index from a column
+ * 
+ * @throws cudf::logic_error if `index` is not within the range`[0, input.size())`
+ * 
+ * @param input Column view to get the element from
+ * @param index Index into `input` to get the element at
+ * @param mr Optional, The resource to use for all returned allocations
+ * @return std::unique_ptr<scalar> Scalar containing the single value
+ */
+std::unique_ptr<scalar> get_element(
```

Comment:
I commented on the reductions PR (that PR didn't mention it depends on this one).   Why isn't this a method on device_view. Can't we just add a `column_view::element()` method to parallel the current `column_device_view::elem ent()`?  

In any case, I don't think copying.hpp is the right place for this?

Reply:
I think the reason for column_view arg is dictionary specialization. We need to get key(), indices() from `dictionary_column_view`. (we can't get these from `device_view`).

Reply:
Reason for keeping it in copying.hpp: `get_element` is getting single element from source which is similar to `copy_range`. 
is there better place than copying.hpp? 
Alternatively, get_element could become a member function of column_view.

Reply:
I did not mean to get these from device_view. I mean to mimic the element() API of device_view.  Why can't you have an `element()` API in `column_view_base` implemented in `column_view`, `dictionary_column_view`, etc?

Reply:
if it is `column_view` member method, should it return `T` or `scalar`?

Reply:
Which was requested in the feature  request?

Reply:
a Scalar